services/core/shared/core/compat/sdk_v5/blocks.js

```diff
@@ -148,7 +148,12 @@ const isQueryFromIndex = params => {
 	const isDirectQuery = ['id', 'height', 'heightBetween'].some(prop => paramProps.includes(prop));
 
 	const sortOrder = params.sort ? params.sort.split(':')[1] : undefined;
-	const isLatestBlockFetch = (paramProps.length === 3 && params.limit === 1 && params.offset === 0 && sortOrder === 'desc');
+	const isLatestBlockFetch = (paramProps.length === 1 && params.limit === 1)
+		|| (paramProps.length === 2
+			&& ((params.limit === 1 && params.offset === 0)
+				|| (sortOrder === 'desc' && (params.limit === 1 || params.offset === 0))
+			))
+		|| (paramProps.length === 3 && params.limit === 1 && params.offset === 0 && sortOrder === 'desc');
```

Comment:
Can you please remind me why is it so important to determine whether is it the last block or not?
What `isLatestBlockFetch` literally means? Latest block to fetch?

Reply:
`isLatestBlockFetch` evaluates the request params that determine the conditions on which the latest forged block is to be fetched from Lisk Core. If unclear, we could rename it to `isFetchLastForgedBlockFromCore` or to something else. I'm open to recommendations.

Without this check, we would always end up querying our `blocksDB` index. Thus, we would never update the index with the newly forged blocks from Lisk Core. This was the issue because of which our index wasn't getting updated automatically even though `performLastBlockUpdate` was executed everytime the `newBlock` event (signal) was fired.